File: flipped_material/cpp/code/guessing_game/guessing_game5.c
#include <stdio.h>

int makeGuess(min_value, max_value){
   int stillGuessing = 1;
   int guess;
   while (stillGuessing) {
     printf("Pick a number between %d and %d\n", min_value, max_value);
     scanf("%d", &guess);
     if (guess >= min_value && guess <= max_value){
       stillGuessing = 0;
     }
   }
   return guess;
}

int main(void) {
  int answer = 5;
  int guess = -1;
  int min_value = 1;
  int max_value = 10;
  
  while (answer != guess){
    guess = makeGuess(min_value, max_value);
    
    if (guess < answer){
      printf("Your guess was too low!\n");
    } else if (guess > answer) {
      printf("Your guess was too high!\n");
    }
    
  }
  printf("You won!\n");
  return 0;
}

File: .old/2022/flipped_material/python/code/guessing_game/guessing_game_5.py
from random import randint

def make_guess(min_value, max_value, moves):
    still_guessing = True
    while still_guessing:
        guess = int(input(f"Guess a number between {min_value} and {max_value} "))
        if guess > min_value or guess < max_value:
            still_guessing = False
    return guess, moves + 1

def check_for_hints(guess, answer):
    if guess < answer:
        return "your guess was too low!"
    elif guess > answer:
        return "your guess was too high!"
    return f"{guess} was the right answer. "

min_value = 1
max_value = 100
answer = randint(min_value, max_value)
guess = -1
moves = 0

while answer != guess:
    guess, moves = make_guess(min_value, max_value, moves)
    print(check_for_hints(guess, answer))

print(f"You won in {moves} moves.")
